### PR TITLE
Added SUPPORTS_LEGACYIFEND for DELPHI XE3+

### PR DESCRIPTION
**Pull request by dompiotr85**
At the XE4 release, the Delphi compilers were changed to accept either `$IFEND` or `$ENDIF` to close `$IF` statements. Before XE4, only `$IFEND` could be used to close `$IF` statements. The `$LEGACYIFEND` directive allows you to restore the old behaviour, which is useful if your code is emitting E2029 related to nested `$IF` and `$IFDEF` statements.

My pull request is adding `SUPPORTS_LEGACYIFEND` feature directive that can be used for simple detection of this feature and using it in code.

### CODE REVIEW

**Comment by obones** [comment]
I think this was added by XE3 as mentioned in various messages, like this one:
http://codeverge.com/embarcadero.delphi.non-tech/delphi-compiler-changes-for-xe3/1086718

File: jedi.inc [hunk 1]
```diff
@@ -584,6 +584,7 @@
   HAS_UNIT_VCL_THEMES            Unit Vcl.Themes is available (D16+)
   HAS_UNIT_UXTHEME               Unit (Vcl.)UxTheme is available (D7+)
   HAS_EXCEPTION_STACKTRACE       Exception class has the StackTrace propery (D12+)
+  SUPPORTS_LEGACYIFEND           The compiler supports legacy IFEND (D18+)
```
Comment (obones) [inline jedi.inc]: The message should really be "Compiler supports the LEGACYIFEND directive" to be more similar to other messages above

**Comment by dompiotr85** [comment]
> 
> 
> I think this was added by XE3 as mentioned in various messages, like this one:
> http://codeverge.com/embarcadero.delphi.non-tech/delphi-compiler-changes-for-xe3/1086718

You're right. I was sure that the `{$LEGACYIFEND}` directive was added to the Delphi XE4+ versions because I only considered the entry in the official documentation on the Embarcadero site (http://docwiki.embarcadero.com/RADStudio/Tokyo/en/Legacy_IFEND_(Delphi)).

Changes in my fork version added.

**Comment by obones** [comment]
Thanks, this has now been merged

**Pull request merged**